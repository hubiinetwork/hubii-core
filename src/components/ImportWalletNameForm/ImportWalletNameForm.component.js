import * as React from 'react';
import { Form, Icon } from 'antd';
import {
  Flex,
  Between,
  IconDiv,
  LeftArrow,
  CreateButton,
  WidthEighty,
  Image,
  StyledModalFormLabel
} from './ImportWalletNameForm.style';
import { ModalFormInput, ModalFormItem } from '../ui/Modal';
class ImportWalletNameForm extends React.Component {
  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <div>
        <Between>
          <Flex>
            <LeftArrow
              type="arrow-left"
              onClick={() => console.log('onGoBack function')}
            />
            <span>Importing {this.props.wallet.value} Wallet</span>
          </Flex>
          <div>
            <CreateButton>
              <Icon type="plus" />Create new wallet
            </CreateButton>
          </div>
        </Between>
        <IconDiv>
          <Image src={this.props.wallet.src} />
        </IconDiv>
        <Form
          onSubmit={this.handleSubmit}
          layout="vertical"
          style={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center'
          }}
        >
          <WidthEighty>
            <ModalFormItem
              label={
                <StyledModalFormLabel>
                  Give your Wallet a Name
                </StyledModalFormLabel>
              }
            >
              {getFieldDecorator('Name', {
                rules: [
                  {
                    message: 'Name is required.',
                    required: true
                  }
                ]
              })(<ModalFormInput />)}
            </ModalFormItem>
            <ModalFormItem
              label={
                <StyledModalFormLabel>Set a password</StyledModalFormLabel>
              }
            >
              {getFieldDecorator('password', {
                rules: [
                  {
                    message: 'password is required.',
                    required: true
                  }
                ]
              })(<ModalFormInput />)}
            </ModalFormItem>
            <ModalFormItem
              label={
                <StyledModalFormLabel>Repeat password</StyledModalFormLabel>
              }
            >
              {getFieldDecorator('repeatPassword', {
                rules: [
                  {
                    message: 'Repeat password is required.',
                    required: true
                  }
                ]
              })(<ModalFormInput />)}
            </ModalFormItem>
          </WidthEighty>
        </Form>
      </div>
    );
  }
}
export default Form.create()(ImportWalletNameForm);
