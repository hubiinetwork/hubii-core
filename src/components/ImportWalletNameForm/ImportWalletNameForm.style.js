import styled from 'styled-components';
import Button from '../ui/Button';
import { Icon } from 'antd';
import { ModalFormLabel } from '../ui/Modal';

export const Between = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const CreateButton = styled(Button)`
  background-color: transparent;
  color: ${({ theme }) => theme.palette.light};
  &:hover {
    background-color: ${({ theme }) => theme.palette.info2};
    color: ${({ theme }) => theme.palette.light};
    border-color: ${({ theme }) => theme.palette.light};
  }
`;

export const LeftArrow = styled(Icon)`
  font-size: 20px;
  margin-right: 7px;
  cursor: pointer;
`;

export const Image = styled.img`
  width: 150px;
  height: 50px;
  margin-top: 1rem;
  margin-bottom: 1rem;
`;

export const WidthEighty = styled.div`
  width: 70%;
`;

export const Flex = styled.div`
  display: flex;
  color: ${({ theme }) => theme.palette.light};
`;
export const IconDiv = styled.div`
  display: flex;
  justify-content: center;
`;
export const StyledModalFormLabel = styled(ModalFormLabel)`
  height: 14px;
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
`;
