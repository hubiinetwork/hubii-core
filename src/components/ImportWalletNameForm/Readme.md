Variants of ImportWalletNameForm:

```js
<div style={{ backgroundColor: '#2f4d5c', padding: 25 }}>
  <ImportWalletNameForm
    wallet={{
      src: 'https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png',
      value: 'digitalBitbox1'
    }}
  />
</div>
```
