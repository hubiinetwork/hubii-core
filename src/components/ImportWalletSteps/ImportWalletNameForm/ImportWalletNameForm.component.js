/* eslint-disable */
import * as React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'antd';
import {
  Image,
  IconDiv,
  WidthEighty,
  StyledModalFormLabel,
  ButtonDiv,
  StyledSpan,
  StyledButton,
  StyledBackButton,
} from './ImportWalletNameForm.style';
import { ModalFormInput, ModalFormItem } from '../../ui/Modal';
class ImportWalletNameForm extends React.Component {
  constructor(props) {
    super(props);
    this.handleBack = this.handleBack.bind(this);
    this.handleFinish = this.handleFinish.bind(this);
  }
  handleBack() {
    const { handleBack } = this.props;
    if (handleBack) {
      handleBack();
    }
  }

  handleFinish(e) {
    const { form, handleNext } = this.props;
    e.preventDefault();
    form.validateFields((err, values) => {
      if (!err && handleNext) {
        handleNext(values);
      }
    });
  }
  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <div>
        <IconDiv>
          <Image src={this.props.wallet.src} />
        </IconDiv>
        <Form
          onSubmit={this.handleFinish}
          layout="vertical"
          style={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <WidthEighty>
            <ModalFormItem
              label={
                <StyledModalFormLabel>
                  Give your Wallet a Name
                </StyledModalFormLabel>
              }
            >
              {getFieldDecorator('Name', {
                rules: [
                  {
                    message: 'Name is required.',
                    required: true,
                  },
                ],
              })(<ModalFormInput />)}
            </ModalFormItem>
            <ModalFormItem
              label={
                <StyledModalFormLabel>Set a password</StyledModalFormLabel>
              }
            >
              {getFieldDecorator('password', {
                rules: [
                  {
                    message: 'password is required.',
                    required: true,
                  },
                ],
              })(<ModalFormInput />)}
            </ModalFormItem>
            <ModalFormItem
              label={
                <StyledModalFormLabel>Repeat password</StyledModalFormLabel>
              }
            >
              {getFieldDecorator('repeatPassword', {
                rules: [
                  {
                    message: 'Repeat password is required.',
                    required: true,
                  },
                ],
              })(<ModalFormInput />)}
            </ModalFormItem>
            <ButtonDiv>
              <StyledBackButton onClick={this.handleBack}>
                <StyledSpan>Back</StyledSpan>
              </StyledBackButton>
              <StyledButton htmlType="submit">
                <StyledSpan>Finish</StyledSpan>
              </StyledButton>
            </ButtonDiv>
          </WidthEighty>
        </Form>
      </div>
    );
  }
}

ImportWalletNameForm.propTypes = {
  /**
   * Wallet object to be shown.
   */
  wallet: PropTypes.object,
  /**
   * Function to be executed when back button is pressed
   */
  handleBack: PropTypes.func,
  /**
   * Function to be executed when next is clicked.
   */
  handleNext: PropTypes.func,
    /**
   * ant design form
   */
  form: PropTypes.object,
};

export default Form.create()(ImportWalletNameForm);
