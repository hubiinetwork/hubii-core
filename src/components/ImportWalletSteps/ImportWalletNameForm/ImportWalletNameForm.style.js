import styled from 'styled-components';
import Button from 'components/ui/Button';
import { ModalFormLabel } from 'components/ui/Modal';

export const Between = styled.div`
  display: flex;
  align-items: center;
  .ant-form-item-label{
    display: flex;
    align-self: start;
    margin-left: -13px;
  }
`;

export const Image = styled.img`
  width: 150px;
  height: 44px;
  margin-top: 1rem;
  margin-bottom: 1rem;
`;

export const WidthEighty = styled.div`
  width: 70%;
`;

export const Flex = styled.div`
  display: flex;
  color: ${({ theme }) => theme.palette.light};
`;
export const IconDiv = styled.div`
  display: flex;
  justify-content: center;
`;
export const StyledModalFormLabel = styled(ModalFormLabel)`
  height: 14px;
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
`;
export const StyledTitle = styled.span`
  font-size: 16px;
  font-weight: 500;
  line-height: 19px;
`;

export const ButtonDiv = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 50px;
  margin-bottom: 26px;
`;
export const StyledButton = styled(Button)`
  background-color: ${({ disabled: white }) =>
    white && 'transparent !important'};
  font-size: 12px;
  font-weight: 500;
  border-width: 2px;
  height: 40px;
  width: 162px;
  border: ${({ disabled: white, theme }) =>
    white && `2px solid ${theme.palette.secondary4} !important`};
  min-width: ${({ current: width }) => (width === 0 ? '260px' : '190px')};
  color: ${({ disabled: white, theme }) =>
    white
      ? `${theme.palette.secondary4} !important`
      : `${theme.palette.light} !important`};
  &:hover {
    background-color: ${({ disabled: white }) =>
      white && 'transparent !important'};
    border: ${({ disabled: white, theme }) =>
      white && `2px solid ${theme.palette.secondary4} !important`};
  }
`;

export const StyledBackButton = styled(Button)`
  height: 40px;
  width: 70px;
  margin-right: 8px;
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.palette.light};
`;

export const StyledSpan = styled.span`
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  text-align: center;
`;
