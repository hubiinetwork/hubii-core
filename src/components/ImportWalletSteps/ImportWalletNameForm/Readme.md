Variants of ImportWalletNameForm:

```js
<div
  style={{
    backgroundColor: '#2f4d5c',
    padding: 25,
    width: 576,
    height: 576
  }}
>
  <ImportWalletNameForm
    wallet={{
      src: 'https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png',
      value: 'digitalBitbox1'
    }}
    loading={false}
  />
</div>
```

```js
<div
  style={{
    backgroundColor: '#2f4d5c',
    padding: 25,
    width: 576,
    height: 576
  }}
>
  <ImportWalletNameForm
    wallet={{
      src: 'https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png',
      value: 'digitalBitbox1'
    }}
    loading={true}
  />
</div>
```
