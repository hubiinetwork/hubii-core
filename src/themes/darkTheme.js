// ****** white ****** //

const white = '#ffffff';

// ****** teal ****** //

const teal = '#3b6174';
const tealDark = '#263c46';
const tealLight = '#26404d';
const tealLighter = '#395664';
const tealDarker = '#2f4d5c';
const teal50 = 'rgba(47,77,92,.5)';

// ****** seaGreen ****** //

const seaGreen = '#3df5cd';
const seaGreenLight = 'rgba(80,227,194,.05)';
const seaGreen50 = 'rgba(80,227,194,.5)';

// ****** grey ****** //

const grey = 'rgba(255,255,255,0.5)';
const greyDim = 'rgba(255,255,255,0.3)';
const greyLight = '#C0CDD3';
const greyDark = '#43616F';
const greyDarker = '#445f6c';
const grey50 = 'rgba(140,165,177,.5)';
const grey8C = '#8CA5B1';

// ****** black ****** //

const black = '#122b2b';
const black50 = 'rgba(36, 59, 71, 0.5)';
const blackLight = '#27404c';
const blackLighter = '#404f56';

// ****** green ****** //

const green = 'greenyellow';
const greenLight = 'rgba(173, 255, 47, 0.5)';

// ****** yellow ****** //

const orange = 'orange';

// ****** shadows ****** //

const shadowLight = '0 2px 5px 0 rgba(0, 0, 0, 0.15)';
const shadowDark = '0 3px 7.5px 0 rgba(0, 0, 0, 0.15)';

export default {
  iconColor: '',
  mode: 'dark',
  color: 'red',
  shadows: {
    light: shadowLight,
    dark: shadowDark
  },
  palette: {
    primary: teal,
    primary1: tealLight,
    primary2: tealLighter,
    primary3: tealDark,
    primary4: tealDarker,
    primary5: teal50,

    secondary: grey,
    secondary1: greyLight,
    secondary2: greyDark,
    secondary3: greyDarker,
    secondary4: grey50,
    secondary5: greyDim,
    secondary6: grey8C,

    dark: black,
    dark1: blackLight,
    dark2: blackLighter,
    dark3: black50,

    light: white,

    info: seaGreen,
    info1: seaGreenLight,
    info2: seaGreen50,

    success: green,
    success1: greenLight,
    warning: orange
  }
};
