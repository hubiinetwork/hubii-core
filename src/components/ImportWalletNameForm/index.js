import ImportWalletNameForm from './ImportWalletNameForm.component';
export default ImportWalletNameForm;
