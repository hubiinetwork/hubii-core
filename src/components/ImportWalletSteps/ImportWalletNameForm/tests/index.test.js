import React from 'react';
import { shallow, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import darkTheme from '../../../../themes/darkTheme';
import ImportWalletNameForm from '../index';

const wallet = {
  src: 'https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png',
  value: 'digitalBitbox1',
};

describe('<ImportWalletNameForm />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ImportWalletNameForm wallet={wallet} handleBack={() => {}} handleNext={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should have correct props', () => {
    const wrapper = shallow(<ImportWalletNameForm wallet={wallet} handleBack={() => {}} handleNext={() => {}} />);
    const wrapperProps = wrapper.instance().props;
    expect(wrapperProps.wallet).toEqual(wallet);
  });

  it('should not renders any text', () => {
    const wrapper = render(
      <ThemeProvider theme={darkTheme}>
        <ImportWalletNameForm wallet={wallet} handleBack={() => {}} handleNext={() => {}} />
      </ThemeProvider>);
    expect(wrapper.text()).toContain('');
  });
});

