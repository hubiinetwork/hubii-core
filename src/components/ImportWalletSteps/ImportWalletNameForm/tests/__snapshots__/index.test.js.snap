// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportWalletNameForm /> should render correctly when loading is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form(ImportWalletNameForm)
    handleBack={[Function]}
    handleNext={[Function]}
    loading={false}
    wallet={
        Object {
            "src": "https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png",
            "value": "digitalBitbox1",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "form": Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      },
      "handleBack": [Function],
      "handleNext": [Function],
      "loading": false,
      "wallet": Object {
        "src": "https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png",
        "value": "digitalBitbox1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "form": Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        },
        "handleBack": [Function],
        "handleNext": [Function],
        "loading": false,
        "wallet": Object {
          "src": "https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png",
          "value": "digitalBitbox1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ImportWalletNameForm /> should render correctly when loading is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form(ImportWalletNameForm)
    handleBack={[Function]}
    handleNext={[Function]}
    loading={true}
    wallet={
        Object {
            "src": "https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png",
            "value": "digitalBitbox1",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "form": Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      },
      "handleBack": [Function],
      "handleNext": [Function],
      "loading": true,
      "wallet": Object {
        "src": "https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png",
        "value": "digitalBitbox1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "form": Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        },
        "handleBack": [Function],
        "handleNext": [Function],
        "loading": true,
        "wallet": Object {
          "src": "https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png",
          "value": "digitalBitbox1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
